### PR TITLE
mes-8867-AccessibilityProvider

### PR DESCRIPTION
**Pull request by RLCorp**
## Description
Moved some functionality from app component to a new service
## Checklist

- [x] PR title includes the JIRA ticket number
- [x] Branch is rebased against the latest develop
- [x] Code has been tested manually
- [x] PR link added to JIRA ticket
- [ ] One review from each scrum team
- [ ] Squashed commit contains the JIRA ticket number

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -173,35 +175,13 @@ export class AppComponent extends LogoutBasePageComponent implements OnInit {
   onAppResumed = (): void => {
     this.store$.dispatch(AppResumed());
     window.MobileAccessibility.usePreferredTextZoom(true);
-    window.MobileAccessibility.getTextZoom(this.getTextZoomCallback);
+    window.MobileAccessibility.getTextZoom(this.accessibilityService.getTextZoomCallback);
```
Comment (matthew2564) [inline src/app/app.component.ts]: Can this be wrapped up into a function and call in here, see original PR

File: src/app/pages/test-report/components/reverse-diagram-link/__tests__/reverse-diagram-link.spec.ts [hunk 2]
```diff
@@ -80,23 +79,23 @@ describe('ReverseDiagramLinkComponent', () => {
       ],
       providers: [
         { provide: ModalController, useClass: ModalControllerMock },
-        { provide: AppComponent, useClass: MockAppComponent },
+        AccessibilityService,
```
Comment (matthew2564) [inline src/app/pages/test-report/components/reverse-diagram-link/__tests__/reverse-diagram-link.spec.ts]: Can we create and use the mock AccessibilityService

File: src/app/app.component.ts (hunk 1 above)
Comment (RLCorp) [reply src/app/app.component.ts]: addressed in newest commit

File: src/app/pages/test-report/components/reverse-diagram-link/__tests__/reverse-diagram-link.spec.ts (hunk 2 above)
Comment (RLCorp) [reply src/app/pages/test-report/components/reverse-diagram-link/__tests__/reverse-diagram-link.spec.ts]: addressed in newest commit

File: src/app/providers/journal/accessibility.service.spec.ts [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      ],
+      providers: [
+        AccessibilityService,
+      ],
+    });
+
+    accessibilityService = TestBed.inject(AccessibilityService);
+    store$ = TestBed.inject(Store);
+    spyOn(store$, 'dispatch');
+  }));
```
Comment (matthew2564) [inline src/app/providers/journal/accessibility.service.spec.ts]: Most of this is unnecessary, the provider has no dependencies and therefore we have no reason to be defining schema, importing StoreModule/TranslateModule/PipesModule etc.

Comment (RLCorp) [reply src/app/providers/journal/accessibility.service.spec.ts]: Fixed in newest commit

**Review by andrewsetterfield — APPROVED** [review]

**Review by gittins7 — APPROVED** [review]

**Pull request merged**